backend/inspirehep/snow/api.py

```diff
@@ -0,0 +1,571 @@
+from inspire_utils.record import get_value
+
+from .errors import CreateTicketException, EditTicketException, SnowAuthenticationError
+from .proxies import snow_instance
+from .utils import get_from_cache, get_response_result, relogin_if_needed, strip_lines
+
+LOGGER = structlog.getLogger()
+
+
+class SnowTicketAPI:
+    """Implements basic methods to interact with SNOW (CERN implementation)"""
+
+    def __init__(self, url, client_id, secret, auth_url, verify_cert):
+        self.client_id = client_id
+        self.secret = secret
+        self.auth_url = auth_url
+        self.verify_cert = verify_cert
+        self.url = url
+
+    @property
+    def headers(self):
+        return {
+            "Authorization": f"Bearer {self.token}",
+            "Content-Type": "application/json",
+        }
+
+    def login(self):
+        login_payload = {
+            "grant_type": "client_credentials",
+            "audience": "snow-api-sso-protected",
+            "client_id": self.client_id,
+            "client_secret": self.secret,
+        }
+        headers = {
+            "Content-Type": "application/x-www-form-urlencoded",
+        }
+        try:
+            response = requests.post(self.auth_url, data=login_payload, headers=headers)
+            response.raise_for_status()
+            self.token = response.json()["access_token"]
+            return True
+        except requests.exceptions.RequestException:
+            raise SnowAuthenticationError
+
+    @relogin_if_needed
+    def create_ticket(self, endpoint, payload):
+        """Creates a ticket
+        Args:
+            endpoint (str): an endpoint name
+            payload (dict): ticket payload
+        """
+        response = requests.post(
+            f"{self.url}/{endpoint}",
+            data=orjson.dumps(payload),
+            headers=self.headers,
+        )
+        response.raise_for_status()
+        return get_response_result(response)["sys_id"]
+
+    @relogin_if_needed
+    def get_ticket(self, ticket_id, params=None):
+        """Get ticket by id
+        Args:
+            params (str|dict): params to be passed to the request query string
+        """
+        if not params:
+            params = {}
+        ticket_response = requests.get(
+            f"{self.url}/u_request_fulfillment/{ticket_id}",
+            params=params,
+            headers=self.headers,
+        )
+        ticket_response.raise_for_status()
+        return get_response_result(ticket_response)
+
+    @relogin_if_needed
+    def get_functional_category(self, category_id):
+        """Get functional categoty by id
+        Args:
+            category_id: SNOW sys_id of the category
+         Returns:
+             human-readable category name (str)
+             None if the category is not found
+        """
+        try:
+            category = (
+                self.get_functional_categories_from_cache()
+                .get(category_id, {})
+                .get("sys_name")
```

Comment:
We have this type of code in many places, shall we add the logic to `self.get_functional_categories_from_cache` and pass the key to return? we can use something like `get_value` and we don't need the nested `get` with defaults.

Reply:
We do it twice, I didn't want to create a function that will store the same data as `get_functional_categories_from_cache` but in a different data structure. I can use get `get_value` instead to get rid 
of chaining `get`.